src/components/SpiceItem.js

```diff
@@ -3,17 +3,23 @@ import StarRating from './StarRating'
 
 class SpiceItem extends React.Component {
 
+  state = {
+    toggle : false
+  }
+
+
+  toggleHeart = () => this.setState({toggle : !this.state.toggle})
```

Comment:
Nice work on getting this feature done! If you're comfortable with the object version of `setState`, try using the callback version next time you need to update state based on a previous version of state, to follow [React best practices](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous):

```js
toggleHeart = () => {
    this.setState(prevState => ({
        toggle : !this.state.toggle
    }))
}
```